### PR TITLE
Try to remove RIS tags in Wiley abstracts

### PR DESCRIPTION
**Pull request by adam3smith**
@zuphilip -- let me know if you have an idea for something more
elegant. Reported here: https://forums.zotero.org/discussion/61983/error-194384372-nameis-too-long-to-sync/#Item_9

### CODE REVIEW

File: Wiley Online Library.js [hunk 1]
```diff
@@ -1215,6 +1222,66 @@ var testCases = [
 				"seeAlso": []
 			}
 		]
+	},
+	{
+		"type": "web",
+		"url": "http://onlinelibrary.wiley.com.libezproxy2.syr.edu/doi/10.1111/polp.12144/full",
```
Comment (zuphilip) [inline Wiley Online Library.js]: I guess we want the url here w/o your proxy, i.e.

http://onlinelibrary.wiley.com/doi/10.1111/polp.12144/full

?

File: Wiley Online Library.js [hunk 2]
```diff
@@ -229,7 +229,14 @@ function scrapeBibTeX(doc, url, pdfUrl) {
 				'&fileFormat=REFERENCE_MANAGER' +
 				'&hasAbstract=CITATION_AND_ABSTRACT';
 	ZU.doPost(postUrl, body, function(text) {
-		//Z.debug(text)
+		//on at least one occasion, Wiley puts RIS within RIS, thus causing havoc.
```
Comment (zuphilip) [inline Wiley Online Library.js]: You mean, Wiley puts metadata of related article into the RIS of the article itself? I would add "related articles" somewhere in the comment here...

File: Wiley Online Library.js [hunk 3]
```diff
@@ -229,7 +229,14 @@ function scrapeBibTeX(doc, url, pdfUrl) {
...
+		//This is trying to prevent this (see last test).
```
Comment (zuphilip) [inline Wiley Online Library.js]: The "last test" might not be always the same... Maybe, point here directly to the url?

File: Wiley Online Library.js [hunk 4]
```diff
@@ -229,7 +229,14 @@ function scrapeBibTeX(doc, url, pdfUrl) {
...
+		textClean = text.split(/Related Articles\r/);
```
Comment (zuphilip) [inline Wiley Online Library.js]: Are you sure the regex works for ```Related Articles: ```? I see there also a colon and space and sometimes `\n`...

**Comment by zuphilip** [comment]
Okay, I see the problem. This really seems to be something Wiley should fix on their end. The solution you are proposing is short but may be fragile. What happens when the abstract with the "Related Articles" things is in the beginning and more important information would follow?

Another option here could be to take the RIS without the abstract and add the abstract separately. This might be more stable, but maybe also more work to been done...

**Comment by adam3smith** [comment]
Yeah, you've convinced me it's too risky. I've reported to Wiley instead.

**Comment by adam3smith** [comment]
Reopening this because this issue has struck again:
https://forums.zotero.org/discussion/62619/error-with-syncing-report-id-1485905285#latest
Wiley never got back to me (except, ironically, to ask if my issue had been resolved to my satisfaction...)
@zuphilip -- let's strategize how to fix this with the least amount of possible side effects. This took forever to troubleshoot and was frustrating for both the user and us.

**Comment by zuphilip** [comment]
Okay, then I would suggest to take the RIS metadata without the abstract and maybe add the abstract in a second step, i.e.
```diff
	var postUrl = '/documentcitationdownloadformsubmit';
 	var body = 'doi=' + encodeURIComponent(doi) + 
 				'&fileFormat=REFERENCE_MANAGER' +
- 				'&hasAbstract=CITATION_AND_ABSTRACT';
+ 				'&hasAbstract=CITATION';
 	ZU.doPost(postUrl, body, function(text) {
```
and then add something like
```js
			//abstract
			var abstract = doc.getElementById("abstract");
			if (!item.abstractNote && abstract) {
				item.abstractNote = abstract.textContent.replace(/^(Abstract|Summary)/i, '');
			}
```
What do you think?

**Comment by adam3smith** [comment]
I liked the idea a lot, but tried it out on the Wiley tests and there are a whole bunch of problems this introduces (missing linebreaks, duplicate abstracts in various languages, etc.). So I don't thinks that'll work, unfortunately...

**Comment by zuphilip** [comment]
Okay, I see. I would suggest to do the following plan:
 * open the bibtex including the abstract where possibly some unwanted related articles with their metadata are mentioned
 * read the BibTeX-file line-by-line
 * look for some starting trigger (e.g. Related Articles) and remember its position
 * go through the other lines and analyze them with respect to the brackets
    * (start with counter=1)
    * if there is an open bracket then increase some counter
    * if there is an closing bracket then decrease the same counter
    * if the counter is zero then remember the ending position
 * delete the content from the starting position to the ending position
 * call BibTeX translator with this optimized content

( I have to rethink this again, but I am distracted now by some conference talks 😃 ;-)

**Comment by adam3smith** [comment]
@zuphilip -- holding off on this. Wiley claims to have fixed this, though last I checked I was still getting the same error.

**Comment by zuphilip** [comment]
Closing this, because it looks that this is fixed or otherwise superseded. CC @adam3smith

**Pull request closed without merging**